test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -870,6 +870,128 @@ public void DirectoryBuildProps()
             .And.HaveStdOut("Hello from TestName");
     }
 
+    /// <summary>
+    /// Overriding default (implicit) properties of file-based apps via implicit build files.
+    /// </summary>
+    [Fact]
+    public void DefaultProps_DirectoryBuildProps()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.cs"), """
+            Console.WriteLine("Hi");
+            """);
+        File.WriteAllText(Path.Join(testInstance.Path, "Directory.Build.props"), """
+            <Project>
+              <PropertyGroup>
+                <ImplicitUsings>disable</ImplicitUsings>
+              </PropertyGroup>
+            </Project>
+            """);
+
+        new DotnetCommand(Log, "run", "Program.cs")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Fail()
+            // error CS0103: The name 'Console' does not exist in the current context
+            .And.HaveStdOutContaining("error CS0103");
+
+        // Converting to a project should not change the behavior.
+
+        new DotnetCommand(Log, "project", "convert", "Program.cs")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Pass();
+
+        new DotnetCommand(Log, "run")
+            .WithWorkingDirectory(Path.Join(testInstance.Path, "Program"))
+            .Execute()
+            .Should().Fail()
+            // error CS0103: The name 'Console' does not exist in the current context
+            .And.HaveStdOutContaining("error CS0103");
+    }
+
+    /// <summary>
+    /// Overriding default (implicit) properties of file-based apps from custom SDKs.
+    /// </summary>
+    [Fact]
+    public void DefaultProps_CustomSdk()
```

Comment:
@DamianEdwards is the behavior captured by this test conceptually what you want to have working for aspire sdk?

Reply:
I think so, yes. IIUC it's demonstrating a file-based app referencing a custom SDK, and that SDK disables implicit usings, which is something explicitly defaulted to on for file-based apps. We should then be able to achieve the same thing for AOT publishing.

Reply:
Yes, if your SDK wants to customize some properties in its Sdk.props like this, that should work.